src/Patterns.jl

```diff
@@ -0,0 +1,80 @@
+# TODO move this?
+
+struct Pattern{T, D} <: AbstractArray{T, D}
+    img::Array{T, D}
+end
```

Comment:
We should also store the pattern type, for vector backends.

Reply:
I don't know what vector backends would need. I guess for lines it would be the direction, linewidth, spacing and colors?

What about cross hatches? And what user defined patterns?

Reply:
Maybe there should be a `LinePattern` and an `ImagePattern`

Reply:
Yeah, that would be pretty good, I think.  I don't think vector backends would need more than that.